MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,9 +1,187 @@
+            correctAnswer: true),
+        QuizQuestion(
+            image: "Kill Bill",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: true),
+        QuizQuestion(
+            image: "The Avengers",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: true),
+        QuizQuestion(
+            image: "Deadpool",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: true),
+        QuizQuestion(
+            image: "The Green Knight",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: true),
+        QuizQuestion(
+            image: "Old",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false),
+        QuizQuestion(
+            image: "The Ice Age Adventures of Buck Wild",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false),
+        QuizQuestion(
+            image: "Tesla",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false),
+        QuizQuestion(
+            image: "Vivarium",
+            text: "Рейтинг этого фильма больше чем 6?",
+            correctAnswer: false),
+    ]
+
+    private var currentQuestionIndex: Int = 0
+    private var correctAnswersCount: Int = 0
+    private var currentQuestionCorrectAnswer: Bool? {
+        guard let question = questions[safe: currentQuestionIndex] else { return nil }
+        return question.correctAnswer
+    }
+
     // MARK: - Lifecycle
+
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        guard let firstQuestion = questions.first else { return }
+
+        show(quiz: convert(model: firstQuestion))
+    }
+
+    private func checkAnswerCorrectness(for chosenAnswer: Bool) -> Bool {
+        chosenAnswer == currentQuestionCorrectAnswer
+    }
+
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        QuizStepViewModel(
+            image: UIImage(named: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questions.count)")
+    }
+
+    private func show(quiz step: QuizStepViewModel) {
+        imageView.image = step.image
+        textLabel.text = step.question
+        counterLabel.text = step.questionNumber
+    }
+
+    private func show(quiz result: QuizResultsViewModel) {
+        let alert = UIAlertController(
+            title: result.title,
+            message: result.text,
+            preferredStyle: .alert)
+        
+        let action = UIAlertAction(title: result.buttonText, style: .default) { _ in
+            self.currentQuestionIndex = 0
+
+            self.correctAnswersCount = 0
+
+            let firstQuestion = self.questions[self.currentQuestionIndex]
+            let viewModel = self.convert(model: firstQuestion)
+            self.show(quiz: viewModel)
+        }
```

Comment:
Замыкания вы будете рассматривать далее, однако советую уже сейчас использовать слабый список захвата на контроллер `[weak self] in` и обращаться к свойствам и функциям через `self?`

Содержимое действия можно вынести в отдельную функцию. 

```suggestion
        let action = UIAlertAction(title: result.buttonText, style: .default) { [weak self] _ in
            self?.restartQuiz()
        }
```